native/src/binaryRead.cpp

```diff
@@ -2573,18 +2573,41 @@ bool ResultPublisher::publish(FoundMapDataObject o) {
 			// 		newerLiveMap = true;
 			// 	}
 			// }
-			if (ex->points.size() >= r->points.size() || o.zoom >= 13) {
+			if (ex->points.size() > r->points.size() || o.zoom >= 13) {
 				return false;
+			}
+
+			bool replace = false;
+			if (r->points.size() > ex->points.size()) {
+				replace = true;
 			} else {
-				auto it = result.begin();
-				for (; it != result.end(); it++) {
-					if (it->obj == ex) {
-						result.erase(it);
-						break;
-					}
+				double exLength = 0.0;
+				double length = 0.0;
+				for (int pointIdx = 1; pointIdx < r->points.size(); pointIdx++) {
+					const auto& exPrevPoint = ex->points[pointIdx - 1];
+					const auto& exCurrPoint = ex->points[pointIdx];
+					exLength += squareDist31TileMetric(exPrevPoint.first, exPrevPoint.second, exCurrPoint.first, exCurrPoint.second);
+
```

Comment:
This method is very slow, if 2 roads are exactly same - so it's fast check 1 and last point exact coordinates are equals ( for > 15 zoom).
For other zooms we have no choice to merge roads or to display 2 of them.